cartoframes/auth/credentials.py

```diff
@@ -23,79 +23,111 @@ class Credentials(object):
     instantiate a :obj:`Credentials` object.
 
     Args:
-        creds (:obj:`cartoframes.Credentials`, optional): Credentials instance
-        key (str, optional): API key of user's CARTO account
+        api_key (str, optional): API key of user's CARTO account
         username (str, optional): Username of CARTO account
         base_url (str, optional): Base URL used for API calls. This is usually
             of the form `https://eschbacher.carto.com/` for user `eschbacher`.
             On premises installations (and others) have a different URL
             pattern.
-        cred_file (str, optional): Pull credentials from a stored file. If this
-            and all other args are not entered, Credentials will attempt to
-            load a user config credentials file that was previously set with
-            Credentials(...).save().
-
-    Raises:
-        RuntimeError: If not enough credential information is passed and no
-            stored credentials file is found, this error will be raised.
 
     Example:
 
         .. code::
 
             from cartoframes.auth import Credentials, Context
-            creds = Credentials(key='abcdefg', username='eschbacher')
-            con = Context(creds=creds)
-
+            credentials = Credentials(api_key='abcdefg', username='eschbacher')
     """
 
-    def __init__(self, creds=None, key=None, username=None, base_url=None,
-                 cred_file=None, api_key=None):
-        self._key = None
-        self._username = None
-        self._base_url = None
-        key = key or api_key
-        if creds and isinstance(creds, Credentials):
-            self.key(key=creds.key())
-            self.username(username=creds.username())
-            self.base_url(base_url=creds.base_url())
-        elif (key and username) or (key and base_url):
-            self.key(key=key)
-            self.username(username=username)
-            if base_url:
-                self.base_url(base_url=base_url)
-            else:
-                self.base_url(
-                    base_url='https://{}.carto.com/'.format(self._username)
-                )
-        elif cred_file:
-            self._retrieve(cred_file)
-        else:
-            try:
-                self._retrieve(_DEFAULT_PATH)
-            except Exception:
-                raise RuntimeError(
-                    'Could not load CARTO credentials. Try setting them with '
-                    'the `key` and `username` arguments.'
-                )
-        self._norm_creds()
+    def __init__(self, api_key, username=None, base_url=None, session=None):
```

Comment:
I would use:

```py
def __init__(self, username, api_key='default_public', base_url=None, session=None):
```

To allow `Credentials('username')` and `Credentials('username', 'api_key')`

Reply:
- `api_key='default_public'` good idea!
- `username` can not be required because we can work with `base_url` instead.

Reply:
```
def __init__(self, username=None, api_key='default_public', base_url=None, session=None):
```
?

Reply:
:+1:

Reply:
I will do it after https://github.com/CartoDB/cartoframes/pull/851 merge

Reply:
Done